clang/lib/CIR/CodeGen/CIRGenTypes.cpp

```diff
@@ -202,6 +202,21 @@ mlir::Type CIRGenTypes::convertType(QualType type) {
     break;
   }
 
+  case Type::ConstantArray: {
+    const ConstantArrayType *arrTy = cast<ConstantArrayType>(ty);
+    mlir::Type elemTy = convertTypeForMem(arrTy->getElementType());
+
+    if (!builder.isSized(elemTy)) {
+      cgm.errorNYI(SourceLocation(), "unimplemented size for type", type);
```

Comment:
I don't like the error being handled this way. If you're going to report an error about unimplemented types here, the check for the types should be here. If someone were to change the implementation of isSized(), they would have no way to know that this code needed to be updated.